extensions/wikia/PortableInfobox/tests/nodes/NodeDataTest.php

```diff
@@ -39,93 +39,249 @@ public function sourceDataProvider() {
 	}
 
 	/**
-	 * @covers       \Wikia\PortableInfobox\Parser\Nodes\Node::getSourceLabel
-	 * @dataProvider sourceLabelDataProvider
+	 * @covers       \Wikia\PortableInfobox\Parser\Nodes\Node::getSourcesMetadata
+	 * @dataProvider sourcesMetadataDataProvider
 	 *
 	 * @param $markup
 	 * @param $params
 	 * @param $expected
 	 */
-	public function testSourceLabel( $markup, $params, $expected ) {
+	public function testSourcesMetadata( $markup, $params, $expected ) {
 		$node = \Wikia\PortableInfobox\Parser\Nodes\NodeFactory::newFromXML( $markup, $params );
 
-		$this->assertEquals( $expected, $node->getSourceLabel() );
+		$this->assertEquals( $expected, $node->getSourcesMetadata() );
 	}
 
-	public function sourceLabelDataProvider() {
+	public function sourcesMetadataDataProvider() {
 		return [
-			[ '<data source="test"></data>', [ ], ['test' => ''] ],
-			[ '<image source="test"/>', [ ], ['test' => ''] ],
-			[ '<title source="test"/>', [ ], ['test' => ''] ],
-			[ '<data source="test"><default>{{{test}}}</default></data>',
-				[ ], ['test' => ''] ],
-			[ '<data source="test"><label source="test">{{{test}}}</label><default>{{{test}}}</default></data>',
-				[ ], ['test' => '{{{test}}}'] ],
-			[ '<data source="test"><label>testLabel</label></data>', [ ], ['test' => 'testLabel'] ],
-			[ '<data></data>', [ ], [ ] ],
-			[ '<group><data source="test"><label>labelInsideGroup</label></data></group>', [], ['test' =>'labelInsideGroup'] ],
-			[ '<group>' .
-				'<data source="test"><label>labelInsideGroup</label></data>' .
-				'<data source="test2"><label>labelInsideGroup2</label></data>' .
-				'</group>',
-				[], ['test' =>'labelInsideGroup', 'test2' =>'labelInsideGroup2'] ],
-			[ '<group>' .
-				'<title source="title"/>' .
-				'<image source="image"/>' .
-				'<data source="test"><label>labelInsideGroup</label></data>' .
-				'<data source="test2"><label>labelInsideGroup2</label></data>' .
-				'</group>',
-				[], ['test' =>'labelInsideGroup', 'test2' =>'labelInsideGroup2', 'title' => '', 'image' => ''] ],
-			[ '<data source="test"><default>{{{test 2}}}</default></data>', [ ], [ 'test' => '', 'test 2' => '' ] ],
-			[ '<data source="test1"><default>{{#if: {{{test2|}}}| [[{{{test2}}} with some text]] }}</default></data>',
-				[ ], [ 'test1' => '', 'test2' => '' ] ],
-			[ '<data><default>{{#switch: {{{test2|}}}|{{{test3}}}|{{{test4|kdjk|sajdkfj|}}}]] }}</default></data>',
-				[ ], [ 'test2' => '', 'test3' => '', 'test4' => '' ] ],
-			[ '<data source="test1">' .
+			[
+				'<data source="test"></data>',
+				[],
+				[
+					'test' => [
+						'label' => '',
+						'primary' => true
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "primary" 11 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)